econml/dml/dynamic_dml.py

```diff
@@ -0,0 +1,660 @@
+
+    random_state: int, :class:`~numpy.random.mtrand.RandomState` instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If :class:`~numpy.random.mtrand.RandomState` instance, random_state is the random number generator;
+        If None, the random number generator is the :class:`~numpy.random.mtrand.RandomState` instance used
+        by :mod:`np.random<numpy.random>`.
+
+    Examples
+    --------
+    A simple example with default models:
+
+    .. testcode::
+        :hide:
+
+        import numpy as np
+        np.set_printoptions(suppress=True)
+
+    .. testcode::
+
+        from econml.dml import DynamicDML
+
+        np.random.seed(123)
+
+        n_panels = 100  # number of panels
+        n_periods = 3  # number of time periods per panel
+        n = n_panels * n_periods
+        groups = np.repeat(a=np.arange(n_panels), repeats=n_periods, axis=0)
+        X = np.random.normal(size=(n, 1))
+        T = np.random.normal(size=(n, 2))
+        y = np.random.normal(size=(n, ))
+        est = DynamicDML()
+        est.fit(y, T, X=X, W=None, groups=groups, inference="auto")
+
+    >>> est.const_marginal_effect(X[:2])
+    array([[-0.012...,  0.031...,  0.069...,  0.111..., -0.349...,
+        -0.076...],
+       [-0.411..., -0.088...,  0.021..., -0.171..., -0.126... ,
+         0.397...]])
+    >>> est.effect(X[:2], T0=0, T1=1)
+    array([-0.225..., -0.378...])
+    >>> est.effect(X[:2], T0=np.zeros((2, n_periods*T.shape[1])), T1=np.ones((2, n_periods*T.shape[1])))
+    array([-0.225..., -0.378...])
+    >>> est.coef_
+    array([[-0.191...],
+       [-0.057...],
+       [-0.023...],
+       [-0.136...],
+       [ 0.107...],
+       [ 0.227...]])
+    >>> est.coef__interval()
+    (array([[-0.333...],
+        [-0.171...],
+        [-0.158...],
+        [-0.352...],
+        [-0.045...],
+        [ 0.049...]]),
+    array([[-0.050...],
+        [ 0.056...],
+        [ 0.112...],
+        [ 0.079...],
+        [ 0.260...],
+        [ 0.405...]]))
+    """
+
+    def __init__(self, *,
+                 model_y='auto', model_t='auto',
+                 featurizer=None,
+                 fit_cate_intercept=True,
+                 linear_first_stages=False,
+                 discrete_treatment=False,
+                 categories='auto',
+                 cv=2,
+                 mc_iters=None,
+                 mc_agg='mean',
+                 random_state=None):
+        self.fit_cate_intercept = fit_cate_intercept
+        self.linear_first_stages = linear_first_stages
+        self.featurizer = clone(featurizer, safe=False)
+        self.model_y = clone(model_y, safe=False)
+        self.model_t = clone(model_t, safe=False)
+        super().__init__(discrete_treatment=discrete_treatment,
+                         discrete_instrument=False,
+                         categories=categories,
+                         cv=GroupKFold(cv) if isinstance(cv, int) else cv,
```

Comment:
The user can pass either an int or a cv object. If it's an int, we use GroupKFold (because it's a group split). I don't delegate this to `_OrthoLearner` because of some conflict with stratification when `T` is categorical.